src/sysupdate/updatectl.c

```diff
@@ -0,0 +1,1226 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <getopt.h>
+#include <locale.h>
+
+#include "sd-bus.h"
+
+#include "build.h"
+#include "bus-error.h"
+#include "bus-label.h"
+#include "bus-locator.h"
+#include "bus-map-properties.h"
+#include "errno-list.h"
+#include "format-table.h"
+#include "json-util.h"
+#include "main-func.h"
+#include "pager.h"
+#include "pretty-print.h"
+#include "strv.h"
+#include "sysupdate-update-set-flags.h"
+#include "sysupdate-util.h"
+#include "terminal-util.h"
+#include "verbs.h"
+
+static PagerFlags arg_pager_flags = 0;
+static bool arg_legend = true;
+static bool arg_reboot = false;
+static bool arg_offline = false;
+static BusTransport arg_transport = BUS_TRANSPORT_LOCAL;
+static char *arg_host = NULL;
+
+#define SYSUPDATE_TARGET_INTERFACE "org.freedesktop.sysupdate1.Target"
+
+typedef struct Version {
+        char *version;
+        UpdateSetFlags flags;
+        char **changelog;
+        char *contents_json;
+} Version;
+
+static void version_clear(Version *v) {
```

Comment:
we nowadays call "shallow" destructors like this xyz_done(). please follow this rule.